config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/util/environment/EnvironmentConfigurationUtils.java

```diff
@@ -36,40 +46,91 @@
+
+        return entityFileMap;
     }
 
-    private String getEnvValue(String key, Object o) {
-        if (o instanceof String) {
-            return (String) o;
+    /**
+     * Parses the deployment bundle metadata file to extract environmental dependencies.
+     *
+     * @param metaDataFile deployment bundle metadata
+     * @param configFolder config folder path to look for environmental files.
+     * @return pair of bundle file name and environmental entities map.
+     */
+    public Pair<String, Map<String, String>> parseBundleMetadata(File metaDataFile, String configFolder) {
+        if (!metaDataFile.exists()) {
+            throw new MissingEnvironmentException("Metadata file " + metaDataFile.toString() + " does not exist.");
         }
-        if (o instanceof File) {
-            // get the entity type and name
-            String entityType = key.substring(0, key.indexOf('.'));
-            String entityName = key.substring(key.indexOf('.') + 1);
-            return loadConfigFromFile((File) o, entityType, entityName);
+
+        final String envConfigPath = new File(configFolder).getAbsolutePath();
+        final EnvironmentBundleData environmentBundleData;
+        try {
+            final ObjectMapper objectMapper = jsonTools.getObjectMapper(YAML_EXTENSION);
+            objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+            environmentBundleData = objectMapper.readValue(metaDataFile, EnvironmentBundleData.class);
+        } catch (IOException e) {
+            throw new MissingEnvironmentException("Unable to parse the data from metadata file " + metaDataFile.toString(), e);
+        }
+
+        if (environmentBundleData != null && environmentBundleData.getEnvironmentEntities() != null && !environmentBundleData.getEnvironmentEntities().isEmpty()) {
+            final String bundleName = environmentBundleData.getName();
+            final String bundleVersion = environmentBundleData.getVersion();
+            final Map<String, String> environmentValues = new LinkedHashMap<>();
+            final List<LinkedHashMap<String, String>> environmentEntities = environmentBundleData.getEnvironmentEntities();
+            environmentEntities.stream().forEach(environmentEntitiy -> {
+                final String entityType = environmentEntitiy.get("type");
+                final String entityName = !EntityTypes.CLUSTER_PROPERTY_TYPE.equals(entityType) ? environmentEntitiy.get("name") : PREFIX_GATEWAY + environmentEntitiy.get("name");
+                final Pair<String, String> entityFilePair = ENTITY_FILE_MAP.get(entityType);
+                if (null == entityFilePair) {
+                    throw new MissingEnvironmentException("Unexpected entity type " + entityType);
+                }
+
+                final File envConfigFile = new File(envConfigPath, entityFilePair.getRight());
+                environmentValues.put(PREFIX_ENV + entityFilePair.getLeft() + "." + entityName,
```

Comment:
if deployement bundle is already created then we can reuse the Bundle object created earlier
we dont have to reload the config files again

Reply:
deployment bundle contains only mappings of environment entities.